ntbs-service/DataMigration/NotificationMapper.cs

```diff
@@ -229,6 +240,16 @@ private static PatientDetails ExtractPatientDetails(dynamic notification)
             return details;
         }
 
+        private static string RemoveCharactersNotIn(string matchingRegex, string input)
+        {
+            // We assume the matching regex to be of the format "[someLettersHere]+"
```

Comment:
I assume this is where you were considering going regex for regex for regex? I would support the two deep (confirming that the 'matchingRegex' param conforms to the expected format)? If failing regex match probably throwing an ArgumentException or similar?

Reply:
as discussed, regex deemed not worth it given the obscurity of the regex it would require